class G<T>

fun foo(p: <!UNRESOLVED_REFERENCE!>P<!>) {
    val v = <!DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!>p<!> <!USELESS_CAST_STATIC_ASSERT_IS_FINE!>as <!NO_TYPE_ARGUMENTS_ON_RHS!>G?<!><!>
    v!!: G<*>
}