// !DIAGNOSTICS: -NO_VALUE_FOR_PARAMETER
class Tree<T>(<!NOT_A_CLASS, DEPRECATED_UNESCAPED_ANNOTATION!>T<!> element<!SYNTAX!><!>, <!NOT_AN_ANNOTATION_CLASS, DEPRECATED_UNESCAPED_ANNOTATION!>Tree<T><!> left<!SYNTAX!><!>, <!NOT_AN_ANNOTATION_CLASS, DEPRECATED_UNESCAPED_ANNOTATION!>Tree<T><!> right<!SYNTAX!><!>) {}