import <!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>java.lang.Comparable<!> as Comparable

fun f(c: <!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>Comparable<*><!>) {
    <!TYPE_MISMATCH!>c<!>: jet.Comparable<*>
    c: <!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>java.lang.Comparable<*><!>
}