<!UNSUPPORTED_FEATURE!>header<!> fun foo1()
<!UNSUPPORTED_FEATURE!>header<!> val bar1 = <!HEADER_PROPERTY_INITIALIZER!>42<!>
<!UNSUPPORTED_FEATURE!>header<!> class Baz1

<!UNSUPPORTED_FEATURE!>impl<!> fun foo2() = 42
<!MUST_BE_INITIALIZED!><!UNSUPPORTED_FEATURE!>impl<!> val bar2: Int<!>
<!UNSUPPORTED_FEATURE!>impl<!> interface Baz2
