package foo

class ImplicitlyDuplicatedByMiddleAndJs

actual class ActualInMiddleCompatibleInJs
class <!ACTUAL_MISSING!>CompatibleInMiddleActualInJvm<!>

class <!ACTUAL_MISSING!>CompatibleInMiddleAndPlatforms<!>