<!REDUNDANT_MODIFIER_FOR_TARGET!>open<!> interface First

<!REDUNDANT_MODIFIER_FOR_TARGET!>abstract<!> interface Second

<!REDUNDANT_MODIFIER_FOR_TARGET!>final<!> enum class Third {
    FOURTH, 
    FIFTH
}

<!REDUNDANT_MODIFIER_FOR_TARGET!>final<!> object Sixth