class A {
    <!INAPPLICABLE_ANNOTATION!>native constructor() {}<!>
    inner class B {
        <!INAPPLICABLE_ANNOTATION!>native constructor() {}<!>
    }

    <!INAPPLICABLE_ANNOTATION!>native constructor(<!UNUSED_PARAMETER!>x<!>: Int)
<!>}

class C <!INAPPLICABLE_ANNOTATION!>[native]()<!>
