package a

import <!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>java.lang.Deprecated<!> as deprecated
import java.lang.SuppressWarnings as suppresswarnings


<!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>deprecated<!> suppresswarnings val s: String = "";

<!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>deprecated<!> suppresswarnings fun main(args : Array<String>) {
    System.out.println("Hello, world!")
}

class Test(<!PLATFORM_CLASS_MAPPED_TO_KOTLIN!>deprecated<!> val s: String,
           suppresswarnings val x : Int) {}

