annotation class Ann(
        val x1: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Class<*><!>,
        val x2: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Class<in String?><!>,
        val x3: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Class<out String?><!>,
        val x4: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>java.lang.Class<*><!>,
        val x5: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Array<Class<*>><!>,
        val x6: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Array<in Class<*>><!>,
        val x7: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Array<out Class<*>><!>,
        vararg val x8: <!JAVA_LANG_CLASS_PARAMETER_IN_ANNOTATION!>Class<*><!>
)
