enum class MyEnum {
    <!ENUM_ENTRY_USES_DEPRECATED_OR_NO_DELIMITER!>A<!><!SYNTAX!>;<!>
    <!ENUM_ENTRY_USES_DEPRECATED_OR_NO_DELIMITER!>B<!><!SYNTAX!>;<!>
    <!ENUM_ENTRY_USES_DEPRECATED_OR_NO_DELIMITER!>C<!><!SYNTAX!>;<!>
    <!ENUM_ENTRY_USES_DEPRECATED_OR_NO_DELIMITER!>D<!>

    fun foo() = 0
}