// !DIAGNOSTICS: -REDECLARATION

package<!SYNTAX!><!>

fun<!SYNTAX!><!> () {

}

val<!SYNTAX!><!> : Int = 1


class<!SYNTAX!><!> {

}

trait<!SYNTAX!><!> {

}

object <!SYNTAX!><!>{

}

enum class<!SYNTAX!><!> {}

annotation class<!SYNTAX!><!> <!ANNOTATION_CLASS_WITH_BODY!>{}<!>