val <!REDECLARATION!>Test1<!> = null
class <!REDECLARATION!>Test1<!>

val <!REDECLARATION!>Test2<!> = null
interface <!REDECLARATION!>Test2<!>

val <!REDECLARATION!>Test3<!> = null
object <!REDECLARATION!>Test3<!>

val <!REDECLARATION, REDECLARATION!>Test4<!> = null
class <!REDECLARATION, REDECLARATION!>Test4<!>
interface <!REDECLARATION!>Test4<!>
object <!REDECLARATION!>Test4<!>