// !DIAGNOSTICS: -UNUSED_PARAMETER -UNUSED_VARIABLE
annotation class Ann(val arg: Int = 1)

<!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> class A <!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> constructor(<!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> val prop: Int) {
    <!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> val x = 1
    <!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> fun foo(<!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> x: Int) {}

    <!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> object O1 {}
    <!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> companion object O2 {}
}

<!DEPRECATED_UNESCAPED_ANNOTATION!>Ann<!> object O3 {}
